plugin/core/rpc.py

```diff
@@ -246,48 +216,36 @@ def send_error_response(self, request_id: Any, error: Error) -> None:
     def exit(self) -> None:
         self.exiting = True
         self.send_notification(Notification.exit())
-
-    def set_crash_handler(self, handler: Callable) -> None:
-        self._crash_handler = handler
-
-    def set_error_display_handler(self, handler: Callable) -> None:
-        self._error_display_handler = handler
-
-    def set_transport_failure_handler(self, handler: Callable) -> None:
-        self._transport_fail_handler = handler
-
-    def handle_transport_failure(self) -> None:
-        debug('transport failed')
-        self.transport = None
-        if self._transport_fail_handler is not None:
-            self._transport_fail_handler()
-        if self._crash_handler is not None:
-            self._crash_handler()
+        try:
+            self.transport.close()  # type: ignore
+        except AttributeError:
+            pass
 
     def send_payload(self, payload: Dict[str, Any]) -> None:
-        if self.transport:
-            message = format_request(payload)
-            self.transport.send(message)
+        try:
+            self.transport.send(payload)  # type: ignore
+        except AttributeError:
+            pass
```

Comment:
Why do we want to silently ignore those?
Maybe at least print debug message?

Reply:
`self.transport` is set to `None` when the reader writer pipes stop. It means the langserver crashed or we sent a `shutdown` request followed by an `exit` notification. It may happen that we try to send stuff like debounced `textDocument/didChange` or `textDocument/documentHighlight`.